azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -177,10 +177,15 @@ public String getPluginVersion() {
     }
 
     public String getUserAgent() {
-        return String.format("%s/%s %s:%s %s:%s",
-                getPluginName(), getPluginVersion(),
-                INSTALLATION_ID_KEY, getInstallationId(),
-                SESSION_ID_KEY, getSessionId());
+        if (isTelemetryAllowed()) {

```

Comment:
I would prefer `return isTelemetryAllowed ? ... : ... ;`, but leave it up to u.

Reply:
Agree, that will be more concise, will change it later